vm/src/builtins/set.rs

```diff
@@ -997,6 +1029,14 @@ impl AnySet {
     ) -> PyResult<impl std::iter::Iterator<Item = ArgIterable>> {
         Ok(std::iter::once(self.into_iterable(vm)?))
     }
+
+    fn as_inner(&self, _vm: &VirtualMachine) -> PySetInner {
+        match_class!(match self.object.as_object() {
+            ref set @ PySet => set.inner.clone(),
+            ref frozen @ PyFrozenSet => frozen.inner.clone(),
```

Comment:
cloning will result unexpected behaviors when writing.
is it impossible to borrow `&PySetInner` from `self.object`? 
in that case, probably AnySet need to be changed to enum like
```
enum AnySet {
    Set(PyRef<PySet>),
    FrozenSet(PyRef<PyFrozenSet>),
}
```